core/src/utils/bits.rs

```diff
@@ -144,45 +146,43 @@ impl<Store: BitStore, Order: BitOrder> codec::Encode for DecodedBits<Store, Orde
 }
 
 #[doc(hidden)]
-pub struct DecodedBitsVisitor<S, O>(std::marker::PhantomData<(S, O)>);
-impl<Store, Order> scale_decode::Visitor for DecodedBitsVisitor<Store, Order> {
+pub struct DecodedBitsVisitor<S, O, R: TypeResolver>(core::marker::PhantomData<(S, O, R)>);
+
+impl<Store, Order, R: TypeResolver> scale_decode::Visitor for DecodedBitsVisitor<Store, Order, R> {
     type Value<'scale, 'info> = DecodedBits<Store, Order>;
     type Error = scale_decode::Error;
+    type TypeResolver = R;
 
     fn unchecked_decode_as_type<'scale, 'info>(
         self,
         input: &mut &'scale [u8],
-        type_id: scale_decode::visitor::TypeId,
-        types: &'info scale_info::PortableRegistry,
+        type_id: &R::TypeId,
```

Comment:
hmm, what are these changes? 

I thought the TypeResolver stuff wasn't integrated yet and done another PR?

Reply:
I built the subxt-core pr on top of the type-resolver PR because I did not want to resolve the conflicts in the end, where files in subxt are modified but deleted in the `subxt-core` PR.